src/main/java/com/artipie/QuartzScheduler.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/artipie/LICENSE.txt
+ */
+package com.artipie;
+
+import com.artipie.asto.Storage;
+import com.artipie.conda.CondaQuartz;
+import com.artipie.rpm.misc.UncheckedConsumer;
+import com.jcabi.log.Logger;
+import java.util.Collection;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.impl.StdSchedulerFactory;
+
+/**
+ * Quartz scheduler holder: starts scheduler, schedules the jobs and adds hook to shutdown.
+ * @since 0.23
+ */
+public final class QuartzScheduler {
+
+    /**
+     * Abstract storage.
+     */
+    private final Storage asto;
+
+    /**
+     * Repository configurations.
+     */
+    private final Collection<RepoConfig> config;
+
+    /**
+     * Ctor.
+     * @param asto Abstract storage
+     * @param config Repositories
+     */
+    public QuartzScheduler(final Storage asto, final Collection<RepoConfig> config) {
+        this.asto = asto;
+        this.config = config;
+    }
+
+    /**
+     * Start scheduler, add shutdown hook and schedule the jobs.
+     */
+    void setup() {
```

Comment:
Maybe `start` is a better name for this method? (because it actually starts scheduler)